### ISSUE TITLE
Pelemay building process doesn't work when mix firmware

### ISSUE THREAD

**Issue by zacky1972** [zeam-vm/pelemay#115]
**Describe the bug**

When `mix firmware` on Nerves, Pelemay building process doesn't work.
Therefore, what a library of NIFs in  `Application.app_dir(:pelemay, "/priv")` should have be shared a library of the target (typically ARM) be that of the host (typically x86_64). 

**To Reproduce**
1. `mix nerves.new` to make a new project
2. Edit `mix.exs` to use Pelemay
3. `mix deps.get`
4. Create a module using Pelemay such as
```elixir
defmodule Some do
  require Pelemay
  import Pelemay
  defpelemay do
    def func list do
      list |> Enum.map(& &1 * 2)
    end
  end
end
```
5. set `MIX_TARGET` to some architecture
6. `mix firmware` without `iex -S mix`
7. it won't generate any libraries in `Application.app_dir(:pelemay, "/priv")`.

**Expected behavior**
`mix firmware` invokes a Pelemay building process, and generate some libraries and stub interface code in `Application.app_dir(:pelemay, "/priv")`.

**Screenshots**
None

**Desktop (please complete the following information):**
 - Pelemay Version: branch OutputLog
 - Elixir & Erlang/OTP versions (`elixir --version`): 1.10
 - OS (OS name & version, and `uname -a`): Linux
 - Clang versions (`clang -v`): N/A

**Additional context**
None.

**Comment by fhunleth** [comment]
Here's what I did:

1. `mix nerves.new pelemay_test`
2. Added the module `Some` like above and the Pelemay dependency
3. `export MIX_TARGET=rpi0`
4. `mix deps.get`
5. `mix firmware`

Then I looked inside the firmware image to see what would be on the Raspberry Pi Zero.

1. `mix firmware.unpack`
2. `tree pelemay_test-rpi0/rootfs/srv/erlang/lib/pelemay-0.0.6/`

Here's what I see:

```sh
pelemay_test-rpi0/rootfs/srv/erlang/lib/pelemay-0.0.6/
├── ebin
│   ├── Elixir.Analyzer.beam
│   ├── Elixir.Generator.Name.beam
│   ├── Elixir.Mix.Tasks.Pelemay.beam
│   ├── Elixir.Optimizer.beam
│   ├── Elixir.Pelemay.Db.beam
│   ├── Elixir.Pelemay.Generator.Builder.beam
│   ├── Elixir.Pelemay.Generator.Interface.beam
│   ├── Elixir.Pelemay.Generator.Native.Enum.beam
│   ├── Elixir.Pelemay.Generator.Native.EnumString.beam
│   ├── Elixir.Pelemay.Generator.Native.String.beam
│   ├── Elixir.Pelemay.Generator.Native.Util.beam
│   ├── Elixir.Pelemay.Generator.Native.beam
│   ├── Elixir.Pelemay.Generator.beam
│   ├── Elixir.Pelemay.beam
│   ├── Elixir.SumMag.beam
│   └── pelemay.app
└── priv
    ├── libnifelixirsome.c
    ├── libnifelixirsome.so
    └── pelemay_nif_elixir_some.ex
```

Then I looked at the `.so`:

```sh
$ file pelemay_test-rpi0/rootfs/srv/erlang/lib/pelemay-0.0.6/priv/libnifelixirsome.so
pelemay_test-rpi0/rootfs/srv/erlang/lib/pelemay-0.0.6/priv/libnifelixirsome.so: ELF 32-bit LSB shared object, ARM, EABI5 version 1 (SYSV), dynamically linked, stripped
```

I didn't run this, but this seemed like a pretty good result. Right?

**Comment by zacky1972** [comment]
Thank you!

BTW, This issue was solved by version 0.0.7.
